x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -593,6 +600,37 @@ class PackagePolicyService implements PackagePolicyServiceInterface {
           ...ingestErrorToResponseOptions(error),
         });
       }
+    };
+
+    // Doing in chunks to not overWhelm kibana, if there are too many to delete
```

Comment:
We use p-map with the `concurrency` option in Fleet for this, do you think you can make the change?